src/PhoneNumberValue.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\I18n\PhoneNumber;
+
+use Laminas\I18n\PhoneNumber\Exception\InvalidPhoneNumberException;
+use Laminas\I18n\PhoneNumber\Exception\UnrecognizableNumberException;
+use libphonenumber\NumberParseException;
+use libphonenumber\PhoneNumber as LibPhoneNumber;
+use libphonenumber\PhoneNumberFormat;
+use libphonenumber\PhoneNumberType;
+use libphonenumber\PhoneNumberUtil;
+use libphonenumber\ShortNumberInfo;
+
+use function array_key_exists;
+
+final class PhoneNumberValue
```

Comment:
Should this value object be marked as internal?

Reply:
If you are not sure about this API being `public`, then yes, otherwise you will have a BC problem after `1.0.0`

Reply:
I'm undecided about this - it's a useful tool outside of just validation - also, we don't want psalm disallowing the use of the constants for consumers because they'll be used to construct masks of acceptable types.

I don't like `__toString()` because it contains formatting logic. Perhaps it would be better to store the input string and use that as the `__toString` value. Thoughts?